### DISCUSSION TITLE
How to turn off auto-complete AI assistance *in* the assistant panel?

### DISCUSSION THREAD

**Discussion by xpe** [zed-industries/zed#13328] [Feature Requests]
I'm getting auto-completed AI "assistance" *in* the assistant panel. This snake-eating-its-own-tail behavior is quite unwelcome.

Here is an example... yikes.

<img width="646" alt="image" src="https://github.com/zed-industries/zed/assets/4576329/e72092bc-02fc-46fc-a6c7-c481bc3c3cf2">

**Comment by financelurker** [comment]
I'll stick with vim...

**Reply by xpe** [reply]
not helpful

**Reply by neiltalap** [reply]
You will waste too much time with Vim

**Comment by bajrangCoder** [comment]
Try out this, but I guess it will disable whole inline autocompletion 

```json
"features": {
  "inline_completion_provider": "none"
},
```

**Reply by tumugide** [reply]
This doesn't work.

**Reply by mtfamb** [reply]
Thanks

**Reply by notpeter** [reply]
New setting:
```json
  "edit_predictions": {
    "enabled_in_text_threads": false
  }
```

**Reply by mtfamb** [reply]
> New setting:
> 
> ```json
>   "edit_predictions": {
>     "enabled_in_text_threads": false
>   }
> ```

the other way also works

**Comment by notpeter** [comment]
Add this to your settings:
```json
  "languages": {
    "Markdown": {
      "show_inline_completions": false
    }
  },
```

**Reply by xpe** [reply]
Thanks for the idea, but this is still a hack/workaround. Scoping this to Markdown is too general. Perhaps Zed has a way to scope settings _to the assistant panel itself_? (If not, perhaps it should.)

**Reply by SomeoneToIgnore** [reply]
`"languages"` is a set of overrides rather then any unique setting fields, so you can put `"show_inline_completions"` to the root of the settings to get it globally working.

Just like it's shown in the default settings: 
https://github.com/zed-industries/zed/blob/b701eab44f0728d90d2d65b3bba20263e16897ad/assets/settings/default.json#L183-L185
There's a `zed: open default settings` command to open this json.

**Reply by notpeter** [reply]
> Thanks for the idea, but this is still a hack/workaround. Scoping this to Markdown is too general. Perhaps Zed has a way to scope settings _to the assistant panel itself_? (If not, perhaps it should.)

I'm with you.  I don't want Copilot/Supermaven suggesting what questions I ask Claude.
I offered this as a workaround because at the moment this is the only way I know of to disable Assistant completions without disabling them everywhere.

**Reply by jraedisch** [reply]
Does not seem to work anymore in "classic" text threads.

**Reply by notpeter** [reply]
```json
  "edit_predictions": {
    "enabled_in_text_threads": false
  }
```

**Reply by injust** [reply]
@notpeter Seems like `edit_predictions_disabled_in` would be a better place for this (e.g. a `text_thread` scope).

**Reply by notpeter** [reply]
Yeah, this is included in channel notes for an upcoming settings reorg:
- https://zed.dev/channel/settings-categorization-24770/notes

**Reply by bgub** [reply]
I'm a bit confused... what's the difference between "show_edit_predictions" and "show_completions_on_input"?

**Reply by xpe** [reply]
@bgub I agree these are confusing which is why I suggested renaming them in December 2024 with #22372 -- there is a detailed write up there that might help.

**Comment by PointerToNextPole** [comment]
I saw a solution from https://www.reddit.com/r/ZedEditor/comments/1dh27aj/how_to_disable_auto_complete_in_zed_editor/, just try with the config below:
```json
"show_edit_predictions": false,
"show_completions_on_input": false
```
and it works on my Mac

**Comment by chocolate-brownie** [comment]
Hey, as of 2025 the proper fix for this would look like this in your `settings.json` file

```
  "features": {
    "edit_prediction_provider": "none"
  },

```

**Reply by nikumadev** [reply]
Thank you, this one really works.

**Comment by JeelRajodiya** [comment]
If you have copilot enabled, There is an option to disable the completations just like we can in vscode. 

<img width="410" height="470" alt="image" src="https://github.com/user-attachments/assets/5dbde774-bc94-411c-b9f1-7e78d069c282" />

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by maxwoolfdev** [comment]
This worked for me

```json
"edit_predictions": {
    "provider": "none",
  },
```
